processor/src/operations/io_ops.rs

```diff
@@ -92,10 +91,11 @@ impl Process {
         // build the word in memory order (reverse of stack order)
         let word = [self.stack.get(4), self.stack.get(3), self.stack.get(2), self.stack.get(1)];
 
-        // write the word to memory
-        self.chiplets
-            .memory_mut()
-            .write_word(self.system.ctx(), addr, self.system.clk(), word)?;
+        // write the word to memory and get the previous word
+        let _ = self
+            .chiplets
+            .memory
+            .write_word(self.system.ctx(), addr, self.chiplets.clk, word);
```

Comment:
This silently ignores the potential error comingg from `write_word()`. We should instead do

```rs
self.chiplets
    .memory
    .write_word(self.system.ctx(), addr, self.chiplets.clk, word)?;
```

This will fix the test that's currently failing.